indexer/src/main/java/au/org/aodn/esindexer/utils/VocabsIndexUtils.java

```diff
@@ -24,30 +27,67 @@ public void setVocabService(VocabService vocabService) {
         this.vocabService = vocabService;
     }
 
+    protected ArdcVocabService ardcVocabService;
+    @Autowired
+    public void setArdcVocabService(ArdcVocabService ardcVocabService) {
+        this.ardcVocabService = ardcVocabService;
+    }
+
+    /*
+    The storedResolvedPathCollection is shared between the @PostConstruct method and the @Scheduled method.
+    If the scheduledRefreshVocabsData method runs while init is still processing, there could be concurrency issues.
+    To mitigate this, synchronize access to this shared resource with volatile keyword to ensure proper visibility.
+    The volatile modifier guarantees that any thread that reads a field will see the most recently written value
+     */
+    private volatile Map<String, Map<PathName, String>> storedResolvedPathCollection;
+
     @PostConstruct
     public void init() throws IOException {
         // Check if the initialiseVocabsIndex flag is enabled
         if (initialiseVocabsIndex) {
             log.info("Initialising {} asynchronously", vocabsIndexName);
-            vocabService.populateVocabsDataAsync();
+            storedResolvedPathCollection = ardcVocabService.getResolvedPathCollection();
+            vocabService.populateVocabsDataAsync(storedResolvedPathCollection);
         }
     }
 
     @Scheduled(cron = "0 0 0 * * *")
-    public void scheduledRefreshVocabsData() throws IOException {
-        log.info("Refreshing ARDC vocabularies data");
+    public void scheduledRefreshVocabsData() {
+        try {
+            log.info("Refreshing ARDC vocabularies data");
+            Map<String, Map<PathName, String>> latestResolvedPathCollection = ardcVocabService.getResolvedPathCollection();
 
-        // call synchronous populating method, otherwise existing vocab caches will be emptied while new data hasn't been fully processed yet.
-        vocabService.populateVocabsData();
+            if (!latestResolvedPathCollection.equals(storedResolvedPathCollection)) {
+                log.info("Detected changes in the resolved path collection, updating vocabularies...");
+                vocabService.populateVocabsData(latestResolvedPathCollection);
+                refreshCaches();
 
-        // clear existing caches
-        vocabService.clearParameterVocabCache();
-        vocabService.clearPlatformVocabCache();
-        vocabService.clearOrganisationVocabCache();
+                // update the head if there are new versions
+                synchronized (this) {
+                    storedResolvedPathCollection = latestResolvedPathCollection;
+                    log.info("Updated storedResolvedPathCollection with the latest data.");
+                }
```

Comment:
I attempted to update each vocabulary with the associated specific version, but it seems unnecessary for the purpose of checking if there are version changes from the current context.

This one works with similar logic: upon bean creation, the initial collection of versions is gathered. The cron job will then scan and check if the initial collection of versions has changed. If so, it will make the latest versions the head (update storedResolvedPathCollection), repopulate the vocabularies, and update the associated caches.